core/notifications/src/main/kotlin/com/voximplant/demos/sdk/core/notifications/AudioCallIncomingService.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2011 - 2024, Zingaya, Inc. All rights reserved.
+ */
+
+package com.voximplant.demos.sdk.core.notifications
+
+import android.app.ForegroundServiceStartNotAllowedException
+import android.app.Service
+import android.content.Intent
+import android.content.pm.ServiceInfo
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import androidx.core.app.ServiceCompat
+
+class AudioCallIncomingService : Service() {
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+
+        val id = intent?.getStringExtra("id")
+        val displayName = intent?.getStringExtra("displayName")
+
+        if (id == null) {
+            stopSelf()
+            return START_NOT_STICKY
+        }
+
+        val notification = SystemTrayNotifier(applicationContext).createIncomingCallNotification(id, displayName)
+        if (notification == null) {
+            stopSelf()
+            return START_NOT_STICKY
+        }
+
+        try {
+            ServiceCompat.startForeground(
+                this,
+                1,
+                notification,
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                    ServiceInfo.FOREGROUND_SERVICE_TYPE_SHORT_SERVICE
```

Comment:
according to the android documentation, it is better to implement onTimeout() method that calls stopSelf()

event the incoming call notification is expected to be displayed ~1min maximum, following the recommendations prevents unexpected behavior in corner cases